rust/arrow/src/buffer.rs

```diff
@@ -241,6 +241,67 @@ impl<T: AsRef<[u8]>> From<T> for Buffer {
     }
 }
 
+/// Creating a `Buffer` instance by storing the boolean values into the buffer
+impl std::iter::FromIterator<bool> for Buffer {
+    fn from_iter<I>(iter: I) -> Self
```

Comment:
TODO: forward to the MutableBuffer implementation, since it doesn't look like there will be a performance penalty.